### ISSUE TITLE
Change suggestion: add a whitelist to "camelcase"

### ISSUE THREAD

**Issue by CapacitorSet** [eslint/eslint#7448]
ESLint version: v2.13.1
The problem I want to solve: getting existing libraries which use snake_case to play nice with ESLint
My take on the correct solution to this problem: Having an optional whitelist for the `camelcase` rule, eg. `{"properties": "always", "whitelist": ["foo_bar"]}`

For instance, consider [this line of code](https://github.com/crisbal/Telegram-Bot-Node/blob/es6/src/MasterPlugin.js#L47):

```js
reply({
    type: "text",
    text: `*${plugin.name}* - ${plugin.description}\n\n${plugin.help}`,
    options: {
        parse_mode: "Markdown" // <-- HERE
    }
});
```

In this case, I'm relying on an external library, which eventually relies on [an API which uses snake case](https://core.telegram.org/bots/api#sendmessage); I would like to whitelist all variables and keys called "parse_mode", so that the linter doesn't report any warnings.

**Comment by vitorbal** [comment]
@CapacitorSet thanks for the suggestion! I think what you are looking for is the `{ "properties": "never" }` option of "camelcase": http://eslint.org/docs/rules/camelcase#never

**Comment by CapacitorSet** [comment]
Well, I really should read documentation more in detail.

Thank you for your time.

**Comment by vitorbal** [comment]
@CapacitorSet no problem! Happy to help :)

**Comment by reintroducing** [comment]
@vitorbal I actually just searched google for this same thing. The use case is this: We have a good amount of random variables sprinkled throughout our code which do a back and forth to some APIs that are written in python. Python uses snake_case for variable names and so that we don't have to keep changing variables back and forth after API responses/before sending to our API, we keep those specific variables as snake_case. That being said, we follow camelCase conventions for all of our other JS, so we end up having to put a bunch of eslint disable comments in our code everywhere to not error on the python names. 

It would be really nice to add those specific variables which repeat in a lot of places in our code to a whitelist so we don't have to do this. I don't want to set the property to "never" as a general sweep because then it won't pick up the JS portion of the code we want to keep camelCased.  Does that make sense? Is there any possibility of doing this (other than of course writing a custom rule)?

**Comment by j-f1** [comment]
You could set `properties: never` and do something like this:

```js
callAPI({
  some_variable: someVariable
})
```

**Comment by reintroducing** [comment]
That's not exactly what I'm after, unfortunately. I have a piece of code like this, for instance:

```
const initialState = {
    ...
    monthly_start_date: null, // eslint-disable-line camelcase
    ref_latitude: null, // eslint-disable-line camelcase
    ref_longitude: null, // eslint-disable-line camelcase
    mixpanel_id: null, // eslint-disable-line camelcase
    distance__lt: 1609, // eslint-disable-line camelcase
    distance__gt: null, // eslint-disable-line camelcase
    search_string: null, // eslint-disable-line camelcase
    original_search_string: null, // eslint-disable-line camelcase
    users_latitude: null, // eslint-disable-line camelcase
    users_longitude: null, // eslint-disable-line camelcase
    ...
};
```

Those properties show up in a lot of other files and most of them end up being pushed up to the APIs that way so converting them back and forth is a giant chore everywhere, hence we keep them with the snake_case. It would be nice to whitelist them so that we don't have to do the comments. (and yes, I know I can turn it off on a per file level or per code block level, this is just an example piece of code with a lot of other code in there that I don't want to turn this rule off entirely for).

**Comment by j-f1** [comment]
@reintroducing Those properties shouldn’t be an issue if you set `properties: never` in your ESLint config. Example:

```yaml
rules:
  camelcase:
    - error
    - properties: never
```

**Comment by reintroducing** [comment]
@j-f1 But we have properties that are camelcased as well. I took them out for the example code above, but there are other properties in that object which do have camelCasing (which are JS props, not ones that come from python) and we want to keep those (and all other props in JS objects we set up) to check for camelCasing.
